components/builder-web/app/origin/origin-page/origin-members-tab/origin-members-tab.component.ts

```diff
@@ -119,13 +131,41 @@ export class OriginMembersTabComponent implements OnInit, OnDestroy {
     });
   }
 
+  departFromOrigin(): void {
+    const data = {
+      originName: this.origin.name
+    };
+
+    this.departOriginDialog
+      .open(DepartOriginDialog, { width: '480px', data: data })
+      .afterClosed()
+      .subscribe((confirmed) => {
+        if (confirmed) {
+          this.store.dispatch(departOrigin(this.origin.name, this.token, (originName) => {
+            this.departureRouting(this.isPrivate, originName);
+          }));
+        }
+      });
+  }
+
   submit(username: string) {
     this.store.dispatch(inviteUserToOrigin(username, this.origin.name, this.token));
     const field = this.form.get('username');
     field.setValue('');
     field.markAsPristine();
   }
 
+  private departureRouting(isPrivate: boolean, originName: string): void {
+    // Check to see if the origin has default package visibility of private and route accordingly
+    if (isPrivate) {
+      this.router.navigateByUrl('/origins');
+    } else {
+      this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
+        this.router.navigate(['/origins', originName]);
+      });
+    }
+  }
```

Comment:
I chose to pull the logic for the routing into its own function so it would be easier to follow.  

`this.router.navigateByUrl` causes a page refresh, so when the origin is private, we'll cause a refresh and navigate explicitly to the `/origins` page.

When an origin is public, we'll navigateByUrl to `/` only to force a refresh, and then navigate to `/origins/<origin name>`, right back where we were before but with a different view since now we are no longer a member.